src/instructlab/config/init.py

```diff
@@ -183,6 +210,52 @@ def init(
     )
 
 
+def hw_auto_detect() -> tuple[str | None, int | None, int | None, _train, bool]:
```

Comment:
The signature is pretty vague. When most items can be None you'll have to make a lot of checks later on.
At very least there should be a doc string, explaining exactly what they mean. A dataclass [1] for the result would be a better fit. Since the fields could have names then.

Overall, I would recommend rewriting this, so that returning `None` for most fields is not an option.

[1]https://docs.python.org/3/library/dataclasses.html#module-dataclasses

Reply:
for this first version of auto detection i'd like to opt to keep it this way, this is a high priority feature we want to get in today or tomorrow

Reply:
Temporary solutions like these tend to stick around. At very least you need to provide information about the structure returned. There is no way to infer what those values mean, without looking at the code. Type hints are not enough.

Reply:
made an issue to track this as a follow up #2275